controller/class.tapestry-controller.php

```diff
@@ -59,16 +63,20 @@ public function __construct($postId = 0)
     }
 
     /**
-     * Update Tapestry nodes first then
-     * Update the existing Tapestry if the postId is provided
-     * Otherwise, a new Tapestry will be created
+     * Add A Tapestry
+     * 
+     * Note: the Tapestry should have at max one root node
+     * There should not be any links available at this moment
      * 
      * @param   Object  $tapestry   Tapestry
      * 
      * @return  Object  $tapestry
      */
-    public function updateTapestry($tapestry)
+    public function addTapestry($tapestry)
     {
+        if (isset($tapestry->links) || count($tapestry->nodes) > 1) {
+            return $this->_throwsError('INVALID_TAPESTRY');
+        }
         if (!$this->postId) {
             $this->postId = $this->_updatePost($tapestry, 'tapestry');
         }
```

Comment:
What about the case that there is already a tapestry saved with the passed in ID? You would be adding _additional_ nodes, which is unexpected. Maybe you should check to make sure the tapestry is empty if `$this->postId` is passed in, otherwise return an error.

Reply:
Yeah, I just realize that there will be no such scenario that `$this->postId` would be set when this method is called (the function in `endpoints.php` didn't initialize the postId when calling this method). I'd say it's safe to remove this check.

Reply:
Then, I suggest adding a check to make sure `$this->postId` is not passed in.